language/move-prover/bytecode/src/function_data_builder.rs

```diff
@@ -176,6 +176,12 @@ impl<'env> FunctionDataBuilder<'env> {
         self.mk_bool_call(ast::Operation::Eq, vec![arg1, arg2])
     }
 
+    /// Make an identical equality expression. This is stronger than `make_equal` because
+    /// it requires the exact same representation, not only interpretation.
+    pub fn mk_identical(&self, arg1: Exp, arg2: Exp) -> Exp {
```

Comment:
I didn't understand the motivation for the  new operator "identical".  Could you please explain?

Reply:
`Identical` is used to implement a "let" (via fun_builder.emit_let). In this case you really want to enforce native equality, not just structural equality, which is weaker. Also for $Mutation, you want to have `==` instead the default of equality of the embedded values.

Reply:
Concretely, is Identical being used only for $Mutation?  If not, what else is it being used for?

Reply:
As I said, any place where we model a "let" via an assume. Those lets stem from various points in the bytecode transformation. To see them you can find the function FunctionDataBuilder::emit_let and then let your IDE search for it's usages.

As a side note, the current let implementation assumes loops have been removed. It would require a havoc of the lhs if this will not be eventually maintaned.